lib/JSON/Validator/Joi.pm

```diff
@@ -47,6 +47,9 @@ sub extend {
       for keys %{$self->{properties} || {}};
   }
 
+  push @{ $clone->{required} }, @{ $self->{required} }
+    if $self->{required} && @{$self->{required}};
```

Comment:
I think this should be unshift() and not push().

I also think this only applies if it's an "object".

And the code does not follow the styling in the rest of the file.

Reply:
Ok )